utils/appliance.py

```diff
@@ -77,15 +98,291 @@ def __init__(self, address=None, browser_steal=False):
                 self._url = address.geturl()
         self.browser_steal = browser_steal
         self._db_ssh_client = None
+        # this will be renamed
+        self.sentaku_ctx = sentaku.ImplementationContext(
```

Comment:
I don't like having something called `sentaku_ctx` in Appliance. If we can hold it inside `_context` or `_implementation` or whatever, that would be strongly preferred.